File: DONGUK/프로그래머스_WEEK 3/네트워크.py
from collections import deque

visited = [0] * 201

def search(graph, i):
    q = deque()
    
    q.append(i)
    
    while q:
        top = q.popleft()
        
        visited[top] = 1
        
        for data in graph[top]:
            if visited[data] == 0:
                q.append(data)
    

def solution(n, computers):
    answer = 0
    
    graph = [[]  for _ in range(n+1)]
    
    for i in range(1, n+1):
        for j in range(1, n+1):
            if i != j and computers[i-1][j-1] == 1:
                graph[i].append(j)
    
    
    for i in range(1, n+1):
        if visited[i] == 0:
            answer += 1
            search(graph, i)
            
    return answer

File: JUNHEE/프로그래머스_옛날문제/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

bool is_visit[200] = {false, };

void dfs(vector<vector<int>> computers, int curr, int n) {
    is_visit[curr] = true;
    
    for(int i = 0; i < n; i++){
        if(computers[curr][i] == 1 && !is_visit[i]){
            dfs(computers, i, n);
        }
    }
    
}
int solution(int n, vector<vector<int>> computers) {
    for(int i = 0; i < n; i++){
        if(!is_visit[i]){
            dfs(computers, i, n);
            answer++;
        }
    }
    return answer;
}
